pac4j-core/src/main/java/org/pac4j/core/authorization/generator/FromAttributesAuthorizationGenerator.java

```diff
@@ -1,72 +1,80 @@
 package org.pac4j.core.authorization.generator;
 
+import org.pac4j.core.profile.CommonProfile;
+
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.StringTokenizer;
 
-import org.pac4j.core.profile.CommonProfile;
-
 /**
  * <p>Generate the authorization information by inspecting attributes.</p>
  * <p>The attributes containing the roles separated by the {@link #splitChar} property (can be set through {@link #setSplitChar(String)}) are
  * defined in the constructor. It's the same for the attributes containing the permissions.</p>
- * 
+ *
  * @author Jerome Leleu
  * @since 1.5.0
  */
 public class FromAttributesAuthorizationGenerator<U extends CommonProfile> implements AuthorizationGenerator<U> {
-    
-    private final String[] roleAttributes;
-    
-    private final String[] permissionAttributes;
-    
+
+    private final Iterable<String> roleAttributes;
+
+    private final Iterable<String> permissionAttributes;
+
     private String splitChar = ",";
-    
-    public FromAttributesAuthorizationGenerator(final String[] roleAttributes, final String[] permissionAttributes) {
-        this.roleAttributes = copyArray(roleAttributes);
-        this.permissionAttributes = copyArray(permissionAttributes);
+
+    public FromAttributesAuthorizationGenerator(final Iterable<String> roleAttributes, final Iterable<String> permissionAttributes) {
+        this.roleAttributes = roleAttributes;
+        this.permissionAttributes = permissionAttributes;
     }
 
-    private String[] copyArray(final String[] original) {
-        if (original != null) {
-            return Arrays.copyOf(original, original.length);
+    public FromAttributesAuthorizationGenerator(final String[] roleAttributes, final String[] permissionAttributes) {
+        if (roleAttributes != null) {
+            this.roleAttributes = Arrays.asList(roleAttributes);
```

Comment:
I'm thinking of [this kind](https://www.securecoding.cert.org/confluence/display/java/OBJ06-J.+Defensively+copy+mutable+inputs+and+mutable+internal+components) or [that kind](http://cwe.mitre.org/data/definitions/374) of best practice.

Basically in object oriented programming, the object encapsulates data and if this data is directly modifiable by an external object, then you have a problem. So when you can, you should copy mutable data in prevention, in particular when you don't have responsibility of the code that will call the concerned method. It's a very basic best practice and it was already enforced in the previous code you modify here, so I thought it would be best to continue with that.